app/controllers/base_collision_controller.rb

```diff
@@ -0,0 +1,17 @@
+class BaseCollisionController
+  def initialize(source, destinations)
+    @source       = source
+    @destinations = destinations
+  end
+
+  def tick
+    @destinations.each do |destination|
+      next unless source.rect.intersect_rect?(destination.rect)
+
+      # TODO: Calculate collision point?
+      on_collision(destination)
```

Comment:
Et si on ajoutait `on_collision` sur un `Engine::Model`?
Defaults to `Proc.new { }`

```suggestion
      destination.on_collision(source)
      source.on_collision(destination)
```

Ca implique que c'est au niveau du model que ca se passerait. J'aime bien l'idee d'un controlleur par contre.
Soit on a plusieurs controlleurs qui regardent les collisions, soit on a besoin que d'un seul controlleur et chaque model implemente son callback de collision.

Reply:
J'ai quand meme l'impression qu'il y a beaucoup de state management. C'est pas clair ce qui se retrouve dans le model ou dans le controlleur.

Reply:
Ouais je suis d'accord, c'est pas encore tout à fait clair pour moi non plus, mais j'aime bien l'idée que le model s'occupe uniquement de son state, et expose des actions. Je trouve ça plus simple à réfléchir, à tester, et ça évite que la balle ait à connaitre son environnement pour savoir comment se comporter (par exemple de savoir qu'elle doit faire exploser une brick, qu'elle doit modifier sa vitesse quand elle touche le paddle). Pour moi ça semble plus simple si se sont des rules qui sont décrites à un niveau au dessus du model.